server/src/main/java/org/apache/cassandra/sidecar/config/yaml/InstanceConfigurationImpl.java

```diff
@@ -64,17 +76,21 @@ public class InstanceConfigurationImpl implements InstanceConfiguration
 
     public InstanceConfigurationImpl()
     {
-        this.id = 0;
-        this.host = null;
-        this.port = 9042;
-        this.dataDirs = null;
-        this.stagingDir = null;
-        this.cdcDir = null;
-        this.jmxHost = null;
-        this.jmxPort = 0;
-        this.jmxSslEnabled = false;
-        this.jmxRole = null;
-        this.jmxRolePassword = null;
+        this(0,
+             null,
+             9042,
+             null,
+             null,
+             null,
+             null,
```

Comment:
For commit log, hints and saved caches, can we build it if it is not set. Like we do in Cassandra, by taking in CASSANDRA_HOME

Reply:
I agree that we should simplify Sidecar configuration, where we follow Cassandra's configuration. Ideally, we will only have to configure the `cassandra_home` configuration per instance. And then we resolve the directory files in Cassandra unless explicitly provided.